src/rust/engine/fs/store/src/snapshot.rs

```diff
@@ -85,11 +71,9 @@ impl Snapshot {
       .compat()
       .await?;
 
-    let mut path_stats =
-      Iterator::flatten(path_stats_per_directory.into_iter().map(Vec::into_iter))
-        .collect::<Vec<_>>();
-    #[allow(clippy::unnecessary_sort_by)]
-    path_stats.sort_by(|l, r| l.path().cmp(&r.path()));
+    let path_stats = Iterator::flatten(path_stats_per_directory.into_iter().map(Vec::into_iter))
+      .collect::<Vec<_>>();
+    let path_stats = PathStat::normalize_path_stats(path_stats)?;
```

Comment:
We used to not check for duplicates with this method `from_digest`. Now we do. This might result in more work than we wanted, as I think the stored digest will already be deduped? Lmk if I should revert this.

Reply:
The contents of a `Digest` should already meet the guarantee due to how it is constructed. Probably not worth worrying about, but adding a note saying that it is just an assertion might be good if you end up re-bouncing this off CI.